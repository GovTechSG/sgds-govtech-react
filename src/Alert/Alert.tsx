import classNames from 'classnames';
import * as React from 'react';
import PropTypes from 'prop-types';
//@ts-ignore
import { elementType } from 'prop-types-extra';
import { useUncontrolled } from 'uncontrollable';
import useEventCallback from '@restart/hooks/useEventCallback';
import {
  useBootstrapPrefix,
  SGDSWrapper,
} from '../ThemeProvider/ThemeProvider';
import Fade from '../Fade/Fade';
import CloseButton from '../CloseButton/CloseButton';
import { Variant } from '../utils/types';
import {
  BsPrefixProps,
  BsPrefixRefForwardingComponent,
  TransitionType,
} from '../utils/helpers';
import { AlertLink } from './AlertLink';
import {AlertHeading}  from './AlertHeading';

export interface AlertProps
  extends React.HTMLAttributes<HTMLDivElement>,
    BsPrefixProps {
  /** The Alert visual color variant */
  variant?: Omit<Variant, 'dark'>;
  /** Renders a properly aligned dismiss button, as well as adding extra horizontal padding to the Alert */
  dismissible?: boolean;
  /** Controls the visual state of the Alert */
  show?: boolean;
  /** Callback fired when alert is closed*/
  onClose?: (a: any, b: any) => void;
  /** Sets the aria-label for alert close button*/
  closeLabel?: string;
   /**
   * Animate the alert dismissal. Defaults to using `<Fade>` animation or use
   * `false` to disable. A custom `react-transition-group` Transition can also
   * be provided.
   */
  transition?: TransitionType;
}


const propTypes = {
  /**
   * @default 'alert'
   */
  bsPrefix: PropTypes.string,

  /**
   * The Alert visual variant
   *
   * @type {'primary' | 'secondary' | 'success' | 'danger' | 'warning' | 'info' | 'light'}
   */
  variant: PropTypes.string,

  /**
   * Renders a properly aligned dismiss button, as well as
   * adding extra horizontal padding to the Alert.
   */
  dismissible: PropTypes.bool,

  /**
   * Controls the visual state of the Alert.
   *
   * @controllable onClose
   */
  show: PropTypes.bool,

  /**
   * Callback fired when alert is closed.
   *
   * @controllable show
   */
  onClose: PropTypes.func,

  /**
   * Sets the text for alert close button.
   */
  closeLabel: PropTypes.string,

  /**
   * Animate the alert dismissal. Defaults to using `<Fade>` animation or use
   * `false` to disable. A custom `react-transition-group` Transition can also
   * be provided.
   */
  transition: PropTypes.oneOfType([PropTypes.bool, elementType]),
};

const defaultProps = {
  variant: 'primary',
  show: true,
  transition: Fade,
  closeLabel: 'Close alert',
};

export const Alert: BsPrefixRefForwardingComponent<'div', AlertProps> =
  React.forwardRef<HTMLDivElement, AlertProps>(
    (uncontrolledProps: AlertProps, ref) => {
      const {
        bsPrefix,
        show,
        as: Component = 'div',
        closeLabel,
        className,
        children,
        variant,
        onClose,
        dismissible,
        transition,
        ...props
      } = useUncontrolled(uncontrolledProps, {
        show: 'onClose',
      });

      const prefix = useBootstrapPrefix(bsPrefix, 'alert');
      const handleClose = useEventCallback((e) => {
        if (onClose) {
          onClose(false, e);
        }
      });
      const Transition = transition === true ? Fade : transition;
      const alert = (
        <SGDSWrapper
          as={Component}
          role="alert"
          {...(!Transition ? props : undefined)}
          ref={ref}
          className={classNames(
            className,
            prefix,
            variant && `${prefix}-${variant}`,
            dismissible && `${prefix}-dismissible`
          )}
        >
          {children}
          {dismissible && (
            <CloseButton
              onClick={handleClose}
              aria-label={closeLabel}
              className={`btn-sm`}
            />
          )}
        </SGDSWrapper>
      );

      if (!Transition) return show ? alert : null;

      return (
        <Transition unmountOnExit {...props} ref={undefined} in={show}>
          {alert}
        </Transition>
      );
    }
  );

Alert.displayName = 'Alert';
Alert.defaultProps = defaultProps;
Alert.propTypes = propTypes;

export default Object.assign(Alert, {
  Link: AlertLink,
  Heading: AlertHeading,
});
