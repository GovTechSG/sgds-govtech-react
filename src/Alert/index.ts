export { default as Alert } from './Alert';
export type { AlertProps } from './Alert';

export { default as AlertLink } from './AlertLink';
// export type { AlertLinkProps } from './AlertLink';

export { default as AlertHeading } from './AlertHeading';
