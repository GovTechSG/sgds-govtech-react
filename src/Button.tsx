// import React, { HTMLAttributes, ReactNode } from 'react';

// export interface Props extends HTMLAttributes<HTMLButtonElement> {
// /** Provide a text for the button */
//   children: ReactNode;
// /** Which variant you would like to use */
//   variant: 'primary' | 'secondary';
// }

// /** this is a special button */
// export const Button = ({ children, variant = 'primary', ...props }: Props) => {
//   return (
//     <button
//       {...props}
//       style={{
//         backgroundColor: variant === 'primary' ? 'blue' : 'gray',
//         color: 'white',
//         border: 'none',
//         borderRadius: '100px',
//         padding: '10px',
//         cursor: 'pointer'
//       }}
//     >
//       {children}
//     </button>
//   );
// };
