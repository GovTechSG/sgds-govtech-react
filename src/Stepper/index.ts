export { default as Stepper } from './Stepper'
export { default as useStep } from './useStep'
export type { OnDone, StepMetadata, State as StepperState, UseStepMethods} from './useStep'