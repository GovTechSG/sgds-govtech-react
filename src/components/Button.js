/* eslint-disable jsx-a11y/anchor-is-valid */
import React, { Component } from "react";
import PropTypes from "prop-types";
import getUnhandledProps from "./lib/getUnhandledProps";

class Button extends Component {
  constructor(props) {
    super(props);
    this.state = {
      hovered: false,
      focused: false
    };
  }

  // Handle button hovered
  mouseHoverOn() {
    this.setState({
      hovered: true
    });
  }

  mouseHoverOff() {
    this.setState({
      hovered: false
    });
  }

  getSize() {
    let size = "";
    if (this.props.buttonSize === "small") {
      size = " is-small";
    } else if (this.props.buttonSize === "medium") {
      size = " is-medium";
    } else if (this.props.buttonSize === "large") {
      size = " is-large";
    }
    return size;
  }

  getColorType() {
    if (this.props.colorType === "success") {
      return " is-success";
    }
    if (this.props.colorType === "warning") {
      return " is-warning";
    }

    if (this.props.colorType === "danger") {
      return " is-danger";
    }

    if (this.props.colorType === "info") {
      return " is-info";
    }
    return "";
  }
  // isPrimary, isOutlined, isRounded, disabled, buttonSize
  getClassName() {
    let className = this.props.isSecondary ? "sgds-sec-button" : "sgds-button";
    if (this.props.isPrimary) {
      className = className.concat(" is-primary");
    }
    if (this.props.isOutlined) {
      className = className.concat(" is-outlined");
    }
    if (this.props.isRounded) {
      className = className.concat(" is-rounded");
    }
    if (this.props.isLoading) {
      className = className.concat(" is-loading");
    }
    className = className.concat(this.getSize());
    className = className.concat(this.getColorType());

    return className.concat(` ${this.props.className || ""}`);
  }

  getStyle() {
    if (this.props.isOutlined) {
      return {
        paddingLeft: this.props.paddingHorizontal,
        paddingRight: this.props.paddingHorizontal,
        paddingTop: this.props.paddingVertical,
        paddingBottom: this.props.paddingVertical,
        borderColor: this.props.primaryColor,
        backgroundColor:
          this.state.hovered || this.state.focused
            ? this.props.primaryColor
            : "transparent",
        color:
          this.state.hovered || this.state.focused
            ? "#fff"
            : this.props.primaryColor,
        ...this.props.style
      };
    } else {
      return {
        paddingLeft: this.props.paddingHorizontal,
        paddingRight: this.props.paddingHorizontal,
        paddingTop: this.props.paddingVertical,
        paddingBottom: this.props.paddingVertical,
        backgroundColor: this.props.primaryColor,
        ...this.props.style
      };
    }
  }

  onClick = () => {
    if (!this.props.disabled) {
      this.props.onClick();
    }
  };

  render() {
    const rest = getUnhandledProps(Button.propTypes, this.props);

    return (
      <button
        className={this.getClassName()}
        style={this.getStyle()}
        disabled={this.props.isDisabled}
        onMouseEnter={() =>
          this.setState({
            hovered: true
          })
        }
        onMouseLeave={() =>
          this.setState({
            hovered: false
          })
        }
        onClick={this.onClick}
        onFocus={() => this.setState({ focused: true })}
        onBlur={() => this.setState({ focused: false })}
        {...rest}
      >
        {this.props.children}
      </button>
    );
  }
}

Button.propTypes = {
  className: PropTypes.string,
  style: PropTypes.object,
  isPrimary: PropTypes.bool,
  isLoading: PropTypes.bool,
  isSecondary: PropTypes.bool,
  isOutlined: PropTypes.bool,
  isRounded: PropTypes.bool,
  buttonSize: PropTypes.string,
  isDisabled: PropTypes.bool,
  paddingHorizontal: PropTypes.number,
  paddingVertical: PropTypes.number,
  primaryColor: PropTypes.string,
  onClick: PropTypes.func,
  colorType: PropTypes.string
};

Button.defaultProps = {
  onClick() {}
};

export default Button;
