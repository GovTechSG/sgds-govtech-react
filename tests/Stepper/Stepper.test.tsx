import {
   fireEvent,
   render,
   waitFor,
} from '@testing-library/react';
import { Button } from '../../src/Button';
import { Alert } from '../../src/Alert';
import { Stepper } from '../../src/Stepper';
import { useStep , OnDone } from '../../src/Stepper';
import * as React from 'react';

const MinimalMockStepper = () => {
  const minimalStepMethods = useStep([
    {
      component: 'At Step 1',
      title: 'Step 1',
      stepHeader: 'First Step',
    },
    {
      component: 'At Step 2',
      title: 'Step 2',
      stepHeader: 'Second Step',
    },
    {
      component: 'At Step 3',
      title: 'Create project',
      stepHeader: 'Third Step',
    },
  ]);

  return <Stepper methods={minimalStepMethods} />;
};
interface MockProps {
  onNextMock: VoidFunction;
  onPreviousMock: VoidFunction;
  onArrivedMock: VoidFunction;
}
const InteractiveMockStepper = ({
  onNextMock,
  onPreviousMock,
  onArrivedMock,
}: MockProps) => {
  const stepMethods = useStep([
    {
      component: 'At Step 1',
      title: 'Step 1',
      stepHeader: 'First Step',
      onNextStep: async (onDone: OnDone) => {
        onNextMock();
        onDone();
      },
      onPreviousStep: async (onDone: OnDone) => {
        onPreviousMock();
        onDone();
      },
      onArrived: async () => {
        onArrivedMock();
      },
    },
    {
      component: 'At Step 2',
      title: 'Step 2',
      stepHeader: 'Second Step',
      onNextStep: async (onDone: OnDone) => {
        onNextMock();
        onDone();
      },
      onPreviousStep: async (onDone: OnDone) => {
        onPreviousMock();
        onDone();
      },
      onArrived: async () => {
        onArrivedMock();
      },
    },
    {
      component: <Alert>At Step 3</Alert>,
      title: 'Create project',
      stepHeader: 'Third Step',
      onNextStep: async () => {
        onNextMock();
      },
      onPreviousStep: async (onDone: OnDone) => {
        onPreviousMock();
        onDone();
      },
      onArrived: async () => {
        onArrivedMock();
      },
    },
  ]);
  const {
    state: stepState,
    getTitle,
    setStep,
    nextStep,
    prevStep,
    getNextButtonTitle,
    getBackButtonTitle,
    getComponent,
    reset,
  } = stepMethods;
  return (
    <>
      <Stepper methods={stepMethods} />
      <div>{getComponent()}</div>
      <h3>{getTitle()}</h3>
      <p>currentStep: {stepState.currentStep} </p>
      {getBackButtonTitle() && (
        <Button onClick={prevStep} variant="danger">
          {getBackButtonTitle()}
        </Button>
      )}
      <Button onClick={nextStep}> {getNextButtonTitle()}</Button>
      <Button onClick={reset}>Reset</Button>
      <Button onClick={() => setStep(3)} variant="warning">
        Jump to last page
      </Button>
    </>
  );
};

describe('Stepper', () => {
  it('default html structure, with minimal required stepMethods', () => {
    const { container, asFragment, getByText } = render(<MinimalMockStepper />);

    //outer div
    expect(container.tagName).toEqual('DIV');
    const $outerDiv = container.querySelectorAll('div')[0];
    expect($outerDiv.classList).toContain('sgds');
    expect($outerDiv.classList).toContain('stepper');

    //3 child divs
    expect($outerDiv.children.length).toEqual(3);
    // 1st stepper
    expect($outerDiv.children[0].classList).toContain('stepper-item');
    expect($outerDiv.children[0].classList).toContain('is-active');
    expect($outerDiv.children[0].classList).not.toContain('is-completed');
    expect($outerDiv.children[0].classList).not.toContain(
      'stepper__item--clickable'
    );

    // 2nd stepper 
    expect($outerDiv.children[1].classList).toContain('stepper-item');
    expect($outerDiv.children[1].classList).not.toContain('is-active');
    expect($outerDiv.children[1].classList).not.toContain('is-completed');
    expect($outerDiv.children[1].classList).not.toContain(
      'is-clickable'
    );
    // 3rd stepper
    expect($outerDiv.children[2].classList).toContain('stepper-item');
    expect($outerDiv.children[2].classList).not.toContain('is-active');
    expect($outerDiv.children[2].classList).not.toContain('is-completed');
    expect($outerDiv.children[2].classList).not.toContain(
      'is-clickable'
    );

    //step-marker and step details
    expect(container.querySelectorAll('.stepper-marker').length).toEqual(3);
    expect(container.querySelectorAll('.stepper-detail').length).toEqual(3);
    expect(getByText('First Step')).toBeDefined();
    expect(getByText('1')).toBeDefined();
    expect(getByText('Second Step')).toBeDefined();
    expect(getByText('2')).toBeDefined();
    expect(getByText('Third Step')).toBeDefined();
    expect(getByText('3')).toBeDefined();
    //snapshot
    expect(asFragment()).toMatchSnapshot();
  });

  it('Stepper interactions with onClick methods', async () => {
    const onNextMock = jest.fn(() => 'going next');
    const onPreviousMock = jest.fn(() => 'going previous');
    const onArrivedMock = jest.fn(() => 'arrived');
    const {
      asFragment,
      getByText,
      queryByText,
      container,
      getByRole,
      queryByRole,
      getAllByTestId,
    } = render(
      <InteractiveMockStepper
        onNextMock={onNextMock}
        onPreviousMock={onPreviousMock}
        onArrivedMock={onArrivedMock}
      />
    );
    const $outerDiv = container.querySelectorAll('div')[0];
    // step 1 mock calls
    expect(onArrivedMock).toHaveBeenCalledWith();
    expect(onPreviousMock).not.toHaveBeenCalled();
    expect(onNextMock).not.toHaveBeenCalled();
    jest.clearAllMocks();
    expect(asFragment()).toMatchSnapshot();

    //at stepper 1 expect 'Next' button only
    expect(getByText('Next')).toBeDefined();
    expect(getByText('Next').tagName).toEqual('BUTTON');
    expect(queryByText('Back')).toBeNull();

    //at stepper 1 expect component 1
    expect(getByText('At Step 1')).toBeDefined();
    expect(queryByText('At Step 2')).toBeNull();
    expect(queryByText('At Step 3')).toBeNull();

    // at stepper 1, stepper 2 is not active
    expect($outerDiv.children[0].classList).toContain('is-active');
    expect($outerDiv.children[1].classList).not.toContain('is-active');
    expect($outerDiv.children[2].classList).not.toContain('is-active');

    //at stepper 1, step 1 title is defined, currentStep is defined
    expect(getByText('Step 1').tagName).toEqual('H3');
    expect(getByText('currentStep: 1').tagName).toEqual('P');

    const $nextBtn = getByText('Next');
    // go to Step 2
    fireEvent.click($nextBtn);

    await waitFor(() => {
      expect(onNextMock).toHaveBeenCalled();
      expect(onArrivedMock).toHaveBeenCalledWith();
      expect(onPreviousMock).not.toHaveBeenCalled();
      jest.clearAllMocks();

      expect(getByText('Back')).toBeDefined();
      expect($outerDiv.children[0].classList).not.toContain('is-active');
      expect($outerDiv.children[1].classList).toContain('is-active');
      expect($outerDiv.children[2].classList).not.toContain('is-active');
      expect(asFragment()).toMatchSnapshot();
    });
    // go to Step 3
    fireEvent.click($nextBtn);
    await waitFor(() => {
      expect(onNextMock).toHaveBeenCalled();
      expect(onArrivedMock).toHaveBeenCalledWith();
      expect(onPreviousMock).not.toHaveBeenCalled();
      jest.clearAllMocks();

      // alert component passed in step 3
      expect(getByRole('alert')).toBeInTheDocument();
      expect(getByText('Submit')).toBeDefined();
      expect($outerDiv.children[0].classList).not.toContain('is-active');
      expect($outerDiv.children[1].classList).not.toContain('is-active');
      expect($outerDiv.children[2].classList).toContain('is-active');
      expect(asFragment()).toMatchSnapshot();
    });

    //clicking back button to step 2
    const $backBtn = getByText('Back');
    fireEvent.click($backBtn);
    await waitFor(() => {
      expect(onPreviousMock).toHaveBeenCalled();
      expect(onArrivedMock).toHaveBeenCalledWith();
      expect(onNextMock).not.toHaveBeenCalled();
      jest.clearAllMocks();

      expect(queryByRole('alert')).toBeNull();
      expect(queryByText('Submit')).toBeNull();
      expect($outerDiv.children[0].classList).not.toContain('is-active');
      expect($outerDiv.children[1].classList).toContain('is-active');
      expect($outerDiv.children[2].classList).not.toContain('is-active');
      expect(asFragment()).toMatchSnapshot();
    });

    //click stepper to go to step 1
    const $stepMarkerOne = getAllByTestId('sgds-step')[0];
    fireEvent.click($stepMarkerOne);

    await waitFor(() => {
      expect(onPreviousMock).toHaveBeenCalled();
      expect(onArrivedMock).toHaveBeenCalledWith();
      expect(onNextMock).not.toHaveBeenCalled();
      jest.clearAllMocks();

      expect(container.textContent).toContain('At Step 1');
      expect($outerDiv.children[0].classList).toContain('is-active');
      expect($outerDiv.children[1].classList).not.toContain('is-active');
      expect($outerDiv.children[2].classList).not.toContain('is-active');

      expect(asFragment()).toMatchSnapshot();
    });

    // At stepper 1 , cannot click stepMarker 2 or 3 to move forward

    const $stepMarkerTwo = getAllByTestId('sgds-step')[1];
    const $stepMarkerThree = getAllByTestId('sgds-step')[2];

    fireEvent.click($stepMarkerTwo);
    await waitFor(() => {
      expect(onNextMock).not.toHaveBeenCalled();
      expect(onArrivedMock).not.toHaveBeenCalled();
      jest.clearAllMocks();
      expect($outerDiv.children[0].classList).toContain('is-active');
      expect($outerDiv.children[1].classList).not.toContain('is-active');
      expect($outerDiv.children[2].classList).not.toContain('is-active');

      expect(asFragment()).toMatchSnapshot();
    });
    fireEvent.click($stepMarkerThree);
    await waitFor(() => {
      expect(onNextMock).not.toHaveBeenCalled();
      expect(onArrivedMock).not.toHaveBeenCalled();
      jest.clearAllMocks();
      expect($outerDiv.children[0].classList).toContain('is-active');
      expect($outerDiv.children[1].classList).not.toContain('is-active');
      expect($outerDiv.children[2].classList).not.toContain('is-active');

      expect(asFragment()).toMatchSnapshot();
    });

    //click Next
    fireEvent.click($nextBtn);
    await waitFor(() => {
      expect(onNextMock).toHaveBeenCalled();
      expect(onArrivedMock).toHaveBeenCalled();
      jest.clearAllMocks();
      expect($outerDiv.children[0].classList).not.toContain('is-active');
      expect($outerDiv.children[1].classList).toContain('is-active');
      expect($outerDiv.children[2].classList).not.toContain('is-active');

      expect(asFragment()).toMatchSnapshot();
    });

    //onclick reset goes back to step 1
    const $reset = getByText('Reset');
    expect($reset).toBeDefined();
    fireEvent.click($reset);
    await waitFor(() => {
      expect(onArrivedMock).toHaveBeenCalled();
      jest.clearAllMocks();
      expect($outerDiv.children[0].classList).toContain('is-active');
      expect($outerDiv.children[1].classList).not.toContain('is-active');
      expect($outerDiv.children[2].classList).not.toContain('is-active');
    });

    //onclick Jump to last page Button to fast-forward to step 3
    const $jumpBtn = getByText('Jump to last page');
    fireEvent.click($jumpBtn);

    await waitFor(() => {
      expect(onArrivedMock).toHaveBeenCalled();
      expect($outerDiv.children[0].classList).not.toContain('is-active');
      expect($outerDiv.children[1].classList).not.toContain('is-active');
      expect($outerDiv.children[2].classList).toContain('is-active');
    });
  });
});
