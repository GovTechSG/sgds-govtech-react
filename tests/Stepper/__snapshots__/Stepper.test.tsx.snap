// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stepper Stepper interactions with onClick methods 1`] = `
<DocumentFragment>
  <div
    class="stepper sgds"
  >
    <div
      class="stepper-item is-active"
      data-testid="sgds-step"
      tabindex="0"
    >
      <div
        class="stepper-marker"
      >
        1
      </div>
      <div
        class="stepper-detail"
      >
        <p>
          <b>
            First Step
          </b>
        </p>
      </div>
    </div>
    <div
      class="stepper-item "
      data-testid="sgds-step"
      tabindex="0"
    >
      <div
        class="stepper-marker"
      >
        2
      </div>
      <div
        class="stepper-detail"
      >
        <p>
          <b>
            Second Step
          </b>
        </p>
      </div>
    </div>
    <div
      class="stepper-item "
      data-testid="sgds-step"
      tabindex="0"
    >
      <div
        class="stepper-marker"
      >
        3
      </div>
      <div
        class="stepper-detail"
      >
        <p>
          <b>
            Third Step
          </b>
        </p>
      </div>
    </div>
  </div>
  <div>
    At Step 1
  </div>
  <h3>
    Step 1
  </h3>
  <p>
    currentStep: 1 
  </p>
  <button
    class="btn btn-primary sgds"
    type="button"
  >
     Next
  </button>
  <button
    class="btn btn-primary sgds"
    type="button"
  >
    Reset
  </button>
  <button
    class="btn btn-warning sgds"
    type="button"
  >
    Jump to last page
  </button>
</DocumentFragment>
`;

exports[`Stepper Stepper interactions with onClick methods 2`] = `
<DocumentFragment>
  <div
    class="stepper sgds"
  >
    <div
      class="stepper-item is-completed is-clickable"
      data-testid="sgds-step"
      tabindex="0"
    >
      <div
        class="stepper-marker"
      >
        1
      </div>
      <div
        class="stepper-detail"
      >
        <p>
          <b>
            First Step
          </b>
        </p>
      </div>
    </div>
    <div
      class="stepper-item is-active"
      data-testid="sgds-step"
      tabindex="0"
    >
      <div
        class="stepper-marker"
      >
        2
      </div>
      <div
        class="stepper-detail"
      >
        <p>
          <b>
            Second Step
          </b>
        </p>
      </div>
    </div>
    <div
      class="stepper-item "
      data-testid="sgds-step"
      tabindex="0"
    >
      <div
        class="stepper-marker"
      >
        3
      </div>
      <div
        class="stepper-detail"
      >
        <p>
          <b>
            Third Step
          </b>
        </p>
      </div>
    </div>
  </div>
  <div>
    At Step 2
  </div>
  <h3>
    Step 2
  </h3>
  <p>
    currentStep: 2 
  </p>
  <button
    class="btn btn-danger sgds"
    type="button"
  >
    Back
  </button>
  <button
    class="btn btn-primary sgds"
    type="button"
  >
     Next
  </button>
  <button
    class="btn btn-primary sgds"
    type="button"
  >
    Reset
  </button>
  <button
    class="btn btn-warning sgds"
    type="button"
  >
    Jump to last page
  </button>
</DocumentFragment>
`;

exports[`Stepper Stepper interactions with onClick methods 3`] = `
<DocumentFragment>
  <div
    class="stepper sgds"
  >
    <div
      class="stepper-item is-completed is-clickable"
      data-testid="sgds-step"
      tabindex="0"
    >
      <div
        class="stepper-marker"
      >
        1
      </div>
      <div
        class="stepper-detail"
      >
        <p>
          <b>
            First Step
          </b>
        </p>
      </div>
    </div>
    <div
      class="stepper-item is-completed is-clickable"
      data-testid="sgds-step"
      tabindex="0"
    >
      <div
        class="stepper-marker"
      >
        2
      </div>
      <div
        class="stepper-detail"
      >
        <p>
          <b>
            Second Step
          </b>
        </p>
      </div>
    </div>
    <div
      class="stepper-item is-active"
      data-testid="sgds-step"
      tabindex="0"
    >
      <div
        class="stepper-marker"
      >
        3
      </div>
      <div
        class="stepper-detail"
      >
        <p>
          <b>
            Third Step
          </b>
        </p>
      </div>
    </div>
  </div>
  <div>
    <div
      class="fade alert alert-primary show sgds"
      role="alert"
    >
      At Step 3
    </div>
  </div>
  <h3>
    Create project
  </h3>
  <p>
    currentStep: 3 
  </p>
  <button
    class="btn btn-danger sgds"
    type="button"
  >
    Back
  </button>
  <button
    class="btn btn-primary sgds"
    type="button"
  >
     Submit
  </button>
  <button
    class="btn btn-primary sgds"
    type="button"
  >
    Reset
  </button>
  <button
    class="btn btn-warning sgds"
    type="button"
  >
    Jump to last page
  </button>
</DocumentFragment>
`;

exports[`Stepper Stepper interactions with onClick methods 4`] = `
<DocumentFragment>
  <div
    class="stepper sgds"
  >
    <div
      class="stepper-item is-completed is-clickable"
      data-testid="sgds-step"
      tabindex="0"
    >
      <div
        class="stepper-marker"
      >
        1
      </div>
      <div
        class="stepper-detail"
      >
        <p>
          <b>
            First Step
          </b>
        </p>
      </div>
    </div>
    <div
      class="stepper-item is-active"
      data-testid="sgds-step"
      tabindex="0"
    >
      <div
        class="stepper-marker"
      >
        2
      </div>
      <div
        class="stepper-detail"
      >
        <p>
          <b>
            Second Step
          </b>
        </p>
      </div>
    </div>
    <div
      class="stepper-item "
      data-testid="sgds-step"
      tabindex="0"
    >
      <div
        class="stepper-marker"
      >
        3
      </div>
      <div
        class="stepper-detail"
      >
        <p>
          <b>
            Third Step
          </b>
        </p>
      </div>
    </div>
  </div>
  <div>
    At Step 2
  </div>
  <h3>
    Step 2
  </h3>
  <p>
    currentStep: 2 
  </p>
  <button
    class="btn btn-danger sgds"
    type="button"
  >
    Back
  </button>
  <button
    class="btn btn-primary sgds"
    type="button"
  >
     Next
  </button>
  <button
    class="btn btn-primary sgds"
    type="button"
  >
    Reset
  </button>
  <button
    class="btn btn-warning sgds"
    type="button"
  >
    Jump to last page
  </button>
</DocumentFragment>
`;

exports[`Stepper Stepper interactions with onClick methods 5`] = `
<DocumentFragment>
  <div
    class="stepper sgds"
  >
    <div
      class="stepper-item is-active"
      data-testid="sgds-step"
      tabindex="0"
    >
      <div
        class="stepper-marker"
      >
        1
      </div>
      <div
        class="stepper-detail"
      >
        <p>
          <b>
            First Step
          </b>
        </p>
      </div>
    </div>
    <div
      class="stepper-item "
      data-testid="sgds-step"
      tabindex="0"
    >
      <div
        class="stepper-marker"
      >
        2
      </div>
      <div
        class="stepper-detail"
      >
        <p>
          <b>
            Second Step
          </b>
        </p>
      </div>
    </div>
    <div
      class="stepper-item "
      data-testid="sgds-step"
      tabindex="0"
    >
      <div
        class="stepper-marker"
      >
        3
      </div>
      <div
        class="stepper-detail"
      >
        <p>
          <b>
            Third Step
          </b>
        </p>
      </div>
    </div>
  </div>
  <div>
    At Step 1
  </div>
  <h3>
    Step 1
  </h3>
  <p>
    currentStep: 1 
  </p>
  <button
    class="btn btn-primary sgds"
    type="button"
  >
     Next
  </button>
  <button
    class="btn btn-primary sgds"
    type="button"
  >
    Reset
  </button>
  <button
    class="btn btn-warning sgds"
    type="button"
  >
    Jump to last page
  </button>
</DocumentFragment>
`;

exports[`Stepper Stepper interactions with onClick methods 6`] = `
<DocumentFragment>
  <div
    class="stepper sgds"
  >
    <div
      class="stepper-item is-active"
      data-testid="sgds-step"
      tabindex="0"
    >
      <div
        class="stepper-marker"
      >
        1
      </div>
      <div
        class="stepper-detail"
      >
        <p>
          <b>
            First Step
          </b>
        </p>
      </div>
    </div>
    <div
      class="stepper-item "
      data-testid="sgds-step"
      tabindex="0"
    >
      <div
        class="stepper-marker"
      >
        2
      </div>
      <div
        class="stepper-detail"
      >
        <p>
          <b>
            Second Step
          </b>
        </p>
      </div>
    </div>
    <div
      class="stepper-item "
      data-testid="sgds-step"
      tabindex="0"
    >
      <div
        class="stepper-marker"
      >
        3
      </div>
      <div
        class="stepper-detail"
      >
        <p>
          <b>
            Third Step
          </b>
        </p>
      </div>
    </div>
  </div>
  <div>
    At Step 1
  </div>
  <h3>
    Step 1
  </h3>
  <p>
    currentStep: 1 
  </p>
  <button
    class="btn btn-primary sgds"
    type="button"
  >
     Next
  </button>
  <button
    class="btn btn-primary sgds"
    type="button"
  >
    Reset
  </button>
  <button
    class="btn btn-warning sgds"
    type="button"
  >
    Jump to last page
  </button>
</DocumentFragment>
`;

exports[`Stepper Stepper interactions with onClick methods 7`] = `
<DocumentFragment>
  <div
    class="stepper sgds"
  >
    <div
      class="stepper-item is-active"
      data-testid="sgds-step"
      tabindex="0"
    >
      <div
        class="stepper-marker"
      >
        1
      </div>
      <div
        class="stepper-detail"
      >
        <p>
          <b>
            First Step
          </b>
        </p>
      </div>
    </div>
    <div
      class="stepper-item "
      data-testid="sgds-step"
      tabindex="0"
    >
      <div
        class="stepper-marker"
      >
        2
      </div>
      <div
        class="stepper-detail"
      >
        <p>
          <b>
            Second Step
          </b>
        </p>
      </div>
    </div>
    <div
      class="stepper-item "
      data-testid="sgds-step"
      tabindex="0"
    >
      <div
        class="stepper-marker"
      >
        3
      </div>
      <div
        class="stepper-detail"
      >
        <p>
          <b>
            Third Step
          </b>
        </p>
      </div>
    </div>
  </div>
  <div>
    At Step 1
  </div>
  <h3>
    Step 1
  </h3>
  <p>
    currentStep: 1 
  </p>
  <button
    class="btn btn-primary sgds"
    type="button"
  >
     Next
  </button>
  <button
    class="btn btn-primary sgds"
    type="button"
  >
    Reset
  </button>
  <button
    class="btn btn-warning sgds"
    type="button"
  >
    Jump to last page
  </button>
</DocumentFragment>
`;

exports[`Stepper Stepper interactions with onClick methods 8`] = `
<DocumentFragment>
  <div
    class="stepper sgds"
  >
    <div
      class="stepper-item is-completed is-clickable"
      data-testid="sgds-step"
      tabindex="0"
    >
      <div
        class="stepper-marker"
      >
        1
      </div>
      <div
        class="stepper-detail"
      >
        <p>
          <b>
            First Step
          </b>
        </p>
      </div>
    </div>
    <div
      class="stepper-item is-active"
      data-testid="sgds-step"
      tabindex="0"
    >
      <div
        class="stepper-marker"
      >
        2
      </div>
      <div
        class="stepper-detail"
      >
        <p>
          <b>
            Second Step
          </b>
        </p>
      </div>
    </div>
    <div
      class="stepper-item "
      data-testid="sgds-step"
      tabindex="0"
    >
      <div
        class="stepper-marker"
      >
        3
      </div>
      <div
        class="stepper-detail"
      >
        <p>
          <b>
            Third Step
          </b>
        </p>
      </div>
    </div>
  </div>
  <div>
    At Step 2
  </div>
  <h3>
    Step 2
  </h3>
  <p>
    currentStep: 2 
  </p>
  <button
    class="btn btn-danger sgds"
    type="button"
  >
    Back
  </button>
  <button
    class="btn btn-primary sgds"
    type="button"
  >
     Next
  </button>
  <button
    class="btn btn-primary sgds"
    type="button"
  >
    Reset
  </button>
  <button
    class="btn btn-warning sgds"
    type="button"
  >
    Jump to last page
  </button>
</DocumentFragment>
`;

exports[`Stepper default html structure, with minimal required stepMethods 1`] = `
<DocumentFragment>
  <div
    class="stepper sgds"
  >
    <div
      class="stepper-item is-active"
      data-testid="sgds-step"
      tabindex="0"
    >
      <div
        class="stepper-marker"
      >
        1
      </div>
      <div
        class="stepper-detail"
      >
        <p>
          <b>
            First Step
          </b>
        </p>
      </div>
    </div>
    <div
      class="stepper-item "
      data-testid="sgds-step"
      tabindex="0"
    >
      <div
        class="stepper-marker"
      >
        2
      </div>
      <div
        class="stepper-detail"
      >
        <p>
          <b>
            Second Step
          </b>
        </p>
      </div>
    </div>
    <div
      class="stepper-item "
      data-testid="sgds-step"
      tabindex="0"
    >
      <div
        class="stepper-marker"
      >
        3
      </div>
      <div
        class="stepper-detail"
      >
        <p>
          <b>
            Third Step
          </b>
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
